Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -352,6 +352,15 @@ class HTTP1ConnectionProvider {
             self.state.release(connection: connection, closing: closing)
         }
 
+        // We close defensively here: we may have failed to actually close on other codepaths,
+        // or we may be expecting the server to close. In either case, we want our FD back, so
+        // we close now to cover our backs. We don't care about the result: if the channel is
+        // _already_ closed, that's fine by us.
+        if closing {
+            logger.debug("closing connection")
```

Comment:
I'm totally fine with this but the information here is redundant (because we log it above (`metadata: ["ahc-closing": "\(closing)"]`).

Reply:
@Lukasa whilst we're here, do you recall if there is metadata attached to the logger still here? Ie. can this message be correlated with the request that made the close happen? I think that would be good.

Reply:
Yes, it does: we never remove the metadata key from this logger. We create a new logger when a new request is minted.